src/lib/gundb/UserStorageClass.js

```diff
@@ -883,11 +884,24 @@ export class UserStorage {
     logger.debug('updateFeedIndex', { changed, field, newIndex: this.feedIndex })
   }
 
+  writeFeedEvent(event): Promise<FeedEvent> {
+    this.feedIds[event.id] = event
+    AsyncStorage.setItem('GD_feed', JSON.stringify(this.feedIds))
```

Comment:
to stringify whole feed, them write in the storage is non-fast operation.
so, guessing in the feed loading you're loosing in the feed updates.

as AsyncStorage accepts only strings, the most obvious solution is to store event separately e.g.
`GD_feed_<id>: {JSON data as string}`
if you need to keep event order, you could do it in the separate key shared for whole collection e.g. GD_feed_order (or GD_feed_keys/ids)

Then on each writeFeedEvent you'll update just 2 keys with small amount of data which of course will be faster
To read all feed items on initialize you could use Promise.all

Reply:
probably you could google for best practices to store objects, maps and collections in the async storage

Reply:
yeah, i chose this method for now. localstorage is also limited in space, probably need to use indexed db. feed updates happens just 1 at a time, so writing asynchronously to storage shouldnt be a blocker

Reply:
Okay. i think better solution will be to write an universal wrapper which will allow to store large objects/collection via some unified for both platforms (web+native) interface
Under "the hood" it will use IndexedDB on web and some advanced DB such as https://github.com/realm/realm-js on native

Reply:
As current solution is temporary i agreed, let's keep it